airflow/providers/microsoft/azure/hooks/data_factory.py

```diff
@@ -142,10 +154,17 @@ def get_conn(self) -> DataFactoryManagementClient:
 
         conn = self.get_connection(self.conn_id)
         tenant = conn.extra_dejson.get('extra__azure_data_factory__tenantId')
-        subscription_id = conn.extra_dejson.get('extra__azure_data_factory__subscriptionId')
 
-        credential = None
+        try:
```

Comment:
This is precisely why I think MyPy is useful! Getting " A Subscription ID is required to connect to Azure Data Factory." rather than "None error xxxx" is the whole world of difference for the users.